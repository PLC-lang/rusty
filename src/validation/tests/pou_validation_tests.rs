use crate::{validation::tests::parse_and_validate, Diagnostic};

// unsupported return types
#[test]
fn function_no_return_unsupported() {
    // GIVEN FUNCTION with no return type
    // WHEN parse_and_validate is done
    let diagnostics = parse_and_validate("FUNCTION foo VAR_INPUT END_VAR END_FUNCTION");
    // THEN there should be one diagnostic -> missing return type
    assert_eq!(
        diagnostics,
        vec![Diagnostic::function_return_missing((0..43).into())]
    );
}
