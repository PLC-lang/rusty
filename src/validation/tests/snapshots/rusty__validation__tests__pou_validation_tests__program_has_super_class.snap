---
source: src/validation/tests/pou_validation_tests.rs
expression: "&diagnostics"
---
error[E021]: A program cannot use `EXTEND`
  ┌─ <internal>:5:17
  │
5 │         PROGRAM prog EXTENDS cls
  │                 ^^^^ A program cannot use `EXTEND`


